client/src/pages/DriverLocationInfo/DropoffOrganizerPopup.tsx

```diff
@@ -49,9 +104,56 @@ function processDropoffLocationsForTable(
     });
   };
 
+  interface HttpCheckboxProps {
+    checked: boolean;
+    mutationFn: any; //This needs to be generic enough
+    onSuccess: () => void;
+    onError: () => void;
+  }
+  const HttpCheckbox: React.FC<HttpCheckboxProps> = ({
+    checked,
+    mutationFn,
+    onSuccess,
+    onError,
+  }: HttpCheckboxProps) => {
+    const [isChecked, setIsChecked] = useState(checked);
+    const httpRequest = useMutation({
+      mutationFn: mutationFn,
+      onSuccess(data, variables, context) {
+        setIsChecked((prevVal) => !prevVal);
+        onSuccess();
+      },
+      onError(error, variables, context) {
+        console.error(error);
+        onError();
+      },
+    });
+  
+    //UI
+    return (
+      <div className="relative flex h-full items-center justify-center">
+        {httpRequest.status === "loading" ? (
+          <Loading size="xsmall" thickness="thin" />
+        ) : (
+          <RadixCheckbox.Root
+            className="flex h-5 w-5 items-center justify-center rounded border-2 border-newLeafGreen bg-softGrayWhite shadow-md hover:brightness-110"
+            checked={isChecked}
+            onClick={() => httpRequest.mutate()}
+          >
+            <RadixCheckbox.Indicator className="CheckboxIndicator">
+              <img src={check_icon} alt="" />
+            </RadixCheckbox.Indicator>
+          </RadixCheckbox.Root>
+        )}
+      </div>
+    );
+  };
+  
```

Comment:
Unless I'm mistaken, this component is identical to the `HttpCheckbox` component that you are importing. If I were you, I would move that component into the `components` directory and import it in both places from there.

Reply:
You are right. I created the HttpCheckbox component in the components directory.